libvisual/CMakeLists.txt

```diff
@@ -185,14 +193,29 @@ ENDIF()
 
 # Git revision
 FIND_PACKAGE(Git)
-CONFIGURE_FILE(
-  "${PROJECT_SOURCE_DIR}/update-version.sh.in"
-  "${PROJECT_BINARY_DIR}/update-version.sh"
-  @ONLY
-)
-ADD_CUSTOM_TARGET(update-version ALL
-  COMMAND /bin/sh ${PROJECT_BINARY_DIR}/update-version.sh
-)
+
+EXECUTE_PROCESS(COMMAND ${GIT_EXECUTABLE} describe --always --dirty
+  OUTPUT_VARIABLE LV_REVISION OUTPUT_STRIP_TRAILING_WHITESPACE)
+IF(NOT LV_REVISION)
+  SET(LV_REVISION "unknown")
+ENDIF()
+
+IF(EXISTS ${BINARY_DIR}/version.h)
+  FILE(STRINGS ${BINARY_DIR}/version.h OLD_REVISION REGEX "#define LV_REVISION")
```

Comment:
It reads `PROJECT_BINARY_DIR` (not `BINARY_DIR`) further down and from what I tried, `PROJECT_BINARY_DIR` worked for me but with `BINARY_DIR` the file is not found at least for me.

Reply:
I literally don't know how I forgot `PROJECT_`. I even used it right below